### ISSUE TITLE
`ReduceSum` fails only on `CUDAExecutionProvider` for empty array

### ISSUE THREAD

**Issue by Y-T-G** [microsoft/onnxruntime#26041]
### Describe the issue

The following error occurs when running inference with `CUDAExecutionProvider` with `ReduceSum` on empty array:
```
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running ReduceSum node. Name:'/ReduceSum' Status Message: /onnxruntime_src/onnxruntime/core/providers/cuda/reduction/reduction_ops.cc:291 onnxruntime::common::Status onnxruntime::cuda::PrepareForReduce(const onnxruntime::Tensor*, bool, gsl::span<const long int>, PrepareReduceMetadata&, const onnxruntime::TensorShape*) input_dims[axis] != 0 was false. Can't reduce on dim with value of 0 if 'keepdims' is false. Invalid output shape would be produced. input_shape:{0,0}
```

It doesn't occur with `CPUExecutionProvider` and it also occurs even if `keepdims=True` during export. The output shape also does not seem to be invalid because reducing `[0, 0]` shape on first dim should result in `[0]` which is valid because the number of elements are still 0.

### To reproduce

`onnxruntime-gpu==1.22.0`
`onnxruntime==1.22.1`

```python
from torch import nn
class ReduceTest(nn.Module):
    def forward(self, x):
        return x.sum(0, keepdims=True)

import torch
model = ReduceTest()
torch.onnx.export(model, torch.empty((0,0)), "model.onnx", input_names=["input"])

import onnxruntime as ort
sess = ort.InferenceSession("model.onnx", providers=["CUDAExecutionProvider"])
sess.run(None, dict(input=torch.empty((0,0)).numpy()))
```

### Urgency

I had reported this previously on PyTorch GitHub too 4 months ago on a similar issue.
https://github.com/pytorch/pytorch/issues/66200#issuecomment-2849090692

But since `CPUExecutionProvider` works, but `CUDAExecutionProvider` doesn't work, even with `keepdim=True`, maybe it's ONNXRuntime issue.

### Platform

Linux

### OS Version

Ubuntu-22.04

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.22.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

CUDA

### Execution Provider Library Version

CUDA 12.8
